packages/common/src/dto/is-array-of-specialties.decorator.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  ValidatorConstraint,
+  ValidatorConstraintInterface,
+  ValidationArguments,
+} from 'class-validator';
+import { SpecialtyDTO } from '../dto';
+import {
+  getSpecialtiesByStreamId,
+  getSubSpecialtiesBySpecialtyId,
+  Specialty,
+  Subspecialty,
+} from '../helper';
+import { SkillInformationDTO, SubspecialtyDTO } from './skill-information.dto';
+
+enum SpecialtyErrorEnum {
+  SPECIALTY_REQUIRED = 'Specialty is required',
+  INVALID_SPECIALTY = 'Invalid specialty selection',
+  SUBSPECIALTY_REQUIRED = 'Subspecialty is required',
+  INVALID_SUBSPECIALTY = 'Invalid subspecialty selection',
+}
+
+const isValidString = (string: string) => {
+  if (typeof string === 'string' && string.length > 0) {
+    return true;
+  }
+  return false;
+};
+
+const specialtyNotListed = (specialty: SpecialtyDTO, formSpecialties: Specialty[]) => {
+  // validate specialty is including in selected stream specialty list
+  if (!formSpecialties.find(formSpecialty => formSpecialty.id === specialty.id)) {
+    return true;
+  }
+};
+
+const subspecialtyNotListed = (subspecialty: SubspecialtyDTO, formSpecialties: Subspecialty[]) => {
+  if (!formSpecialties.find(formSpecialty => formSpecialty.id === subspecialty.id)) {
+    return true;
+  }
+};
+
+@ValidatorConstraint({ name: 'specialties', async: false })
+export class IsArrayOfSpecialties implements ValidatorConstraintInterface {
```

Comment:
I'm not sure if there is a way to simplify this at all, this is where the majority of the debt from this ticket lays